src/ffi/types.rs

```diff
@@ -231,6 +332,36 @@ map_enum! {
     TELIO_LOG_TRACE = Trace,
 }
 
+map_enum! {
+    AdapterType <=> TelioAdapterType,
+    BoringTun = BoringTun,
+    WireguardGo = WireguardGoTun,
+    LinuxNativeWg = LinuxNativeTun,
+    WindowsNativeWg = WindowsNativeTun
+}
+
+// Deprecated slog crate had 6 levels
+// but log crate has 5, hence reusing top
+// 2 levels
+map_enum! {
+    Level -> TelioLogLevel,
+    Error = Error,
+    Warn = Warning,
+    Info = Info,
+    Debug = Debug,
+    Trace = Trace,
+}
+
+map_enum! {
+    TelioLogLevel -> Level,
+    Critical = Error,
```

Comment:
Can we drop Critical from our interface since we modify it anyway?

Reply:
Done